pkg/builder/TwoSampleTTestBuilder.go

```diff
@@ -0,0 +1,332 @@
+T is the actual value of the t-statistic, dof(degree of freedom) is 0,
+and P is the p-value. The AltHypothesis (for us) is always set to
+LocationDiffers which specifies the alternative hypothesis that
+the locations of the two samples are not equal. This is a
+two-tailed test.
+For reference about Hypothesis testing with P-value look here...
+https://refactoring.guru/design-patterns/builder/go/example
+
+For these analysis we asume for the null hypothesis that the statistic
+that is generated from the "validation data source", which might be thought of as the
+control source population, is the same as the "data source", which might be thought
+of as the experimental source population.
+This builder's goal is to try to demonstrate a likely difference and assign a
+number between -2 and 2 for the P-value result. The positive or negative indicator depends
+on the statistic being examined. A positive indicator is considered "good" and a negative
+indicator is "bad". To determine the sign we take the mean of the experimental
+"data source" and subtract the mean of the control "validation data source".
+For a statistic like RMSE or BIAS a positive difference is "bad" and a negative difference
+is "good" because we want to minimize the error or the bias in the experiment.
+For CSI "Critical Success Index" it would be the opposite because CSI ranges from 0
+which is poor to 1 which is good. A return value of 0 is neutral / insignificant.
+
+A P-value <= 0.01 (for a 99% major threshold) results in a 2. For 0.01 < P-value <= 0.05
+(for a 95% minor threshold) the result is a 1. A P-value greater than the minor threshold
+will cause a return of 0.
+*/
+import (
+	"errors"
+	"fmt"
+	"github.com/aclements/go-moremath/stats"
+	"github.com/go-playground/validator/v10"
+	"log"
+	"strings"
+)
+
+// use a single instance of Validate, it caches struct info
+
+// setters:
+// The goodnessPolarity indicates if this population is positive good (like for TS/CSI)
+// or negative good like for RMSE or BIAS. The null hypothesis is that the populations
+// are identical so a positive difference for an error value is bad
+// and a negative difference is good (less error in the experimental population
+// is good for error). If the statistic is Threat Score / Critical
+// Succes Index then a positive difference is good (0 is the worst 1 is the best)
+// and a negative difference in the experimental population would be bad.
+// This information is determined outside of this builder (the builder doesn't know
+// what parameter combination is being tested) so the builder must be told
+// what the "goodnes polarity" is +1 or -1.
+func (scc *ScorecardCell) SetGoodnessPolarity(polarity GoodnessPolarity) error {
+	errs := validate.Var(polarity, "required,oneof=-1 1")
+	if errs != nil {
+		log.Print(errs)
+		return errors.New(fmt.Sprint("TwoSampleTTestBuilder SetGoodnessPolarity", errs))
+	} else {
+		scc.goodnessPolarity = polarity
+	}
+	return nil // no errors
+}
```

Comment:
This second else statement is unneeded since the code `return`'s if there's an error. Can we rewrite this as:

```suggestion
	if errs != nil {
		log.Print(errs)
		return errors.New(fmt.Sprint("TwoSampleTTestBuilder SetGoodnessPolarity", errs))
	} 
	scc.goodnessPolarity = polarity

	return nil
}
```

It'd be good to update `SetMajorThreshold` and `SetMinorThreshold` with the same changes.